Documentation/ceph-advanced-configuration.md

```diff
@@ -293,6 +294,56 @@ data:
     osd pool default size = 2
 ```
 
+## Custom CSI ceph.conf Settings
+
+> **WARNING**: It is highly recommended to use the default setting that comes with
+> CephCSI and this can only be used when absolutely necessary.
+> The `ceph.conf` be reset back to default values if/when the configurations are no
+> longer necessary.
+
+If the `csi-ceph-conf-override` ConfigMap is created before the cluster is
+started, the CephCSI pods will automatically pick up the settings. If you
+add the settings to the ConfigMap after the cluster has been initialized,
+you can restart the Rook operator pod and wait for Rook to recreate CSI pods
+to take immediate effect.
+
+After the CSI pods are restarted, the new settings should be in effect.
+
+### Example
+
+In this [Example](https://github.com/rook/rook/tree/{{ branchName }}/deploy/csi-ceph-conf-override.yaml) we
+will set the `rbd_validate_pool` to `false` to skip rbd pool validation
+
+> **WARNING**: Modify Ceph settings carefully to avoid modifying the default
+> configuration.
+> Changing the settings could result in unexpected results if used incorrectly.
+
+```console
+cat csi-ceph-conf-override.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: csi-ceph-conf-override
+  namespace: rook-ceph
+data:
+  ceph.conf: |
+    [global]
+    auth_cluster_required = cephx
+    auth_service_required = cephx
+    auth_client_required = cephx
+    rbd_validate_pool = false
+  # keyring is a required key and its value should be empty
```

Comment:
Why should the keyring be empty?

Reply:
removed it now. Mounting configmap as subpath to csi pods.